external-crates/move/crates/move-package/src/compilation/compiled_package.rs

```diff
@@ -520,6 +522,31 @@ impl CompiledPackage {
             }
         }
 
+        if std::env::var("VERIFIED_SOURCE_BUILD").is_ok() {
+            for prebuilt_dep in prebuilt_deps {
+                let package_dir = PathBuf::from(prebuilt_dep.paths[0].as_str());
+                let package_dir = SourcePackageLayout::try_find_root(&package_dir)?;
+                let package_dir = package_dir.join("build");
+                let dep_name = match prebuilt_dep.name {
+                    Some((name, _)) => name,
+                    None => "unnamed dependency".into(),
+                };
+                let module_path = package_dir
+                    .join(dep_name.as_str())
+                    .join(CompiledPackageLayout::CompiledModules.path());
+
+                let compiled_unit_paths = vec![module_path];
+                let compiled_units = find_filenames(&compiled_unit_paths, |path| {
+                    extension_equals(path, MOVE_COMPILED_EXTENSION)
+                });
+                for bytecode_path in compiled_units.unwrap() {
+                    let decoded =
+                        decode_bytecode_file(package_dir.clone(), dep_name, &bytecode_path)?;
+                    deps_compiled_units.push((dep_name, decoded));
```

Comment:
All changes in support of this main change: `deps_compiled_units` is populated with precompiled package bytecode, which is now part of the `CompiledPackage` result of this function. This means:

- the prebuilt bytecode dependencies are saved under `bytecode_modules/dependencies` of the root package
- `sui client publish` finds all the dependent modules when publishing